Bang/GameEvents/CardEffects/States/WaitingBangCardsAfterIndiansState.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Bang.Players;
+using Bang.PlayingCards;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bang.GameEvents.CardEffects.States
+{
+    internal class WaitingBangCardsAfterIndiansState : HandlerState
+    {
+        private readonly Dictionary<Player, HandlerState> victimStates;
+        private readonly Game.Gameplay gameplay;
+
+        private Player currentVictim;
+
+        public WaitingBangCardsAfterIndiansState(Dictionary<Player, HandlerState> victimStates, Game.Gameplay gameplay)
+        {
+            this.victimStates = victimStates;
+            this.gameplay = gameplay;
+        }
+
+        public override HandlerState ApplyCardEffect(Player player, BangGameCard card, Game.Gameplay gameplay)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override HandlerState ApplyReplyAction(Player victim, BangGameCard firstCard, BangGameCard secondCard)
+        {
+            currentVictim = victim;
+            return ApplyReplyAction(firstCard);
+        }
+
+        public override HandlerState ApplyReplyAction(BangGameCard card)
```

Comment:
I don't like implicit dependency between this method and `ApplyReplyAction(Player victim, BangGameCard firstCard, BangGameCard secondCard)`.

If client firstly calls `ApplyReplyAction(BangGameCard card)`, it will be `NullReferenceException`.

I suppose change `HandlerState ApplyReplyAction(BangGameCard card)` signature into `HandlerState ApplyReplyAction(Player player, BangGameCard card)` and pass `currentVictim` as first argument